npm/ng-packs/packages/components/extensible/src/lib/components/extensible-table/extensible-table.component.ts

```diff
@@ -206,20 +206,20 @@ export class ExtensibleTableComponent<R = any> implements OnChanges {
     });
   }
 
-  hasAvailableActions(rowData): boolean {
-    let isActionAvailable = true;
-    this.actionList.toArray().map(action => {
+  hasVisibleActions(rowData: any): boolean {
+    const actions = this.actionList.toArray();
+    const visibleActions = actions.filter(action => {
       const { visible, permission } = action;
-      if (rowData && action) {
-        const visibilityCheck = visible({
-          record: rowData,
-          getInjected: this.getInjected,
-        });
-        const permissionCheck = this.permissionService.getGrantedPolicy(permission);
 
-        isActionAvailable = visibilityCheck && permissionCheck;
+      if (permission && visible) {
```

Comment:
What happens when visible exits but permission not exits ? I guess this logic must be `|| OR` not `&& AND` ? Am I wrong ? 🤔 && condition is valid for 217.Line not for this line

Reply:
I have also tried these options. In this case it either just looks for visibility or suitable permission availability. However, if both visible and permission values are defined, we need to check both. So, we will have three cases here as:

1. Both permission and visible conditions are there
2. There is a visibility condition but no permission condition
3. There is a permission condition but no visibility condition
4. Neither permission nor visible conditions are there

Where I believe we can handle all of them like this:

<img width="538" alt="image" src="https://github.com/abpframework/abp/assets/92928815/e582874d-f224-40fa-a6b3-5a4cfff4ac4d">